libs/sdk-ui-kit/src/Dialog/ShareDialog/test/ShareDialog.test.tsx

```diff
@@ -0,0 +1,244 @@
+            const userRef = userAccessGrantee.user.ref;
+            const wrapper = await createComponent({ currentUserRef: userRef }, [], [], [userAccessGrantee]);
+            expect(isCurrentUserInGrantees(wrapper)).toBe(true);
+        });
+
+        it("should mark current in grantees when is owner", async () => {
+            const sharedObject: ISharedObject = { ...defaultSharedObject, createdBy: workspaceUser };
+            const wrapper = await createComponent({ sharedObject, currentUserRef: workspaceUser.ref });
+            expect(isCurrentUserInGrantees(wrapper)).toBe(true);
+        });
+
+        it("should remove grantee from list and submit callback for empty grantees list status is private", async () => {
+            const granteeSelector = getGranteeSelector(userAccessGrantee.user);
+            const onApply = jest.fn();
+            const expectedPayload: ISharingApplyPayload = {
+                granteesToAdd: [],
+                granteesToDelete: [
+                    {
+                        granteeRef: userAccessGrantee.user.ref,
+                    },
+                ],
+                isUnderStrictControl: false,
+                isLocked: false,
+                shareStatus: "private",
+            };
+
+            const wrapper = await createComponent({ onApply: onApply }, [], [], [userAccessGrantee]);
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove grantee from list and submit callback for empty grantees list status is private and isUnderStrictControl is true ", async () => {
+            const granteeSelector = getGranteeSelector(userAccessGrantee.user);
+            const onApply = jest.fn();
+            const expectedPayload: ISharingApplyPayload = {
+                granteesToAdd: [],
+                granteesToDelete: [
+                    {
+                        granteeRef: userAccessGrantee.user.ref,
+                    },
+                ],
+                isUnderStrictControl: false,
+                isLocked: false,
+                shareStatus: "private",
+            };
+
+            const wrapper = await createComponent({ onApply: onApply }, [], [], [userAccessGrantee]);
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove grantee from list and submit callback when group all present is public and isUnderStrictControl is false", async () => {
```

Comment:
"when object is public and isUnderStrictControl is false)

Reply:
changed